projects/03app_radio_test/main_Rx.c

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file
+ * @defgroup project_Radio    Radio application
+ * @ingroup projects
+ * @brief This is the radio rx code for radio test
+
+ *
+ * @author Raphael Simoes <raphael.simoes@inria.fr>
+ * @copyright Inria, 2024
+ */
+
+#include <nrf.h>
+#include <stdlib.h>
+#include <string.h>
+// Include BSP headers
+#include "uart.h"
+#include "board.h"
+#include "board_config.h"
+#include "gpio.h"
+#include "protocol.h"
+#include "radio.h"
+#include "timer.h"
+
+uint8_t write_crc[3] = "CRC";
+uint8_t uart_end     = '\x00';
+
+#define DB_BUFFER_MAX_BYTES (255U)       ///< Max bytes in UART receive buffer
+#define DB_UART_BAUDRATE    (1000000UL)  ///< UART baudrate used by the gateway
+#if defined(NRF5340_XXAA) && defined(NRF_APPLICATION)
+#define DB_UART_INDEX (1)  ///< Index of UART peripheral to use
+#else
+#define DB_UART_INDEX (0)  ///< Index of UART peripheral to use
+#endif
+
+static void _led1_blink_fast(void) {
+    db_gpio_toggle(&db_led1);
+}
+
+static void _radio_callback_crc(uint8_t *packet, uint8_t length) {
+    memcpy(packet + length, &write_crc, 3);
+    db_uart_write(DB_UART_INDEX, packet, length + 3);
+}
+
+static void _radio_callback(uint8_t *packet, uint8_t length) {
+    uint8_t RSSI = db_radio_rssi();
+    memcpy(packet + length, &RSSI, 1);
+    memcpy(packet + length + 1, &uart_end, 1);
```

Comment:
```suggestion
    packet[length++] = RSSI;
    packet[length++] = uart_end;
```

Note that this is a bit ugly because this code is writing in a buffer that is allocated by the radio driver. So you could have an overflow here. But in your case that is fine because the driver buffer size is 255 and you know you should only receive 1 byte (length = 1)